wayback/_client.py

```diff
@@ -798,6 +799,42 @@ def get_memento(self, url, timestamp=None, mode=Mode.original, *,
             while True:
                 is_memento = 'Memento-Datetime' in response.headers
                 current_url, current_date, current_mode = _utils.memento_url_data(response.url)
+
+                # In view mode, mementos of redirects get served as a normal
+                # web page that displays info about the redirect and then uses
+                # JavaScript to redirect after a delay. This page is also
+                # missing some important memento headers, even though it is only
+                # served when there is an actual memento.
+                # Do our best to detect and handle this case.
+                if (
+                    current_mode == Mode.view.value
+                    and is_memento is False
+                    and response.status_code == 200
+                    and 'x-archive-src' in response.headers
+                    and re.search(r'Got an? HTTP 3\d\d response at crawl time',
+                                  response.text,
+                                  re.IGNORECASE)
+                ):
+                    redirect_match = re.search(r'<a [^>]*href=(["\'])(/web/\d{14}/.*?)\1[\s|>]', response.text)
```

Comment:
I’m a little worried this will pick up some other, incorrect URL on the page. It might be better to look for the specific timestamp that was requested:

```py
(r'<a [^>]*href=(["\'])(/web/'
 + _utils.format_timestamp(current_date)
 + r'/.*?)\1[\s|>]')`
```